### PR TITLE
History clear

### PR DESCRIPTION
**Pull request by Itay4**
Fixes https://github.com/xonsh/xonsh/issues/3033

### CODE REVIEW

**Comment by laloch** [comment]
Thank's for opening this @Itay4.
It has, however, several drawbacks:
- It doesn't work with SQLite history backend at all.
- It doesn't work with JSON backend, if the history buffer has already been flushed to disk.
- It doesn't pass any parameters to the new history object initializer.

File: xonsh/history/main.py [hunk 1]
```diff
@@ -417,5 +420,8 @@ def history_main(
         hf = hist.flush()
         if isinstance(hf, threading.Thread):
             hf.join()
+    elif ns.action == "clear":
+        builtins.__xonsh__.history = builtins.__xonsh__.history.__class__()
+        print("History cleared")
```
Comment (scopatz) [inline xonsh/history/main.py:425]: please add `file=sys.stderr`

File: xonsh/history/main.py [hunk 2]
```diff
@@ -417,5 +420,8 @@ def history_main(
         hf = hist.flush()
         if isinstance(hf, threading.Thread):
             hf.join()
+    elif ns.action == "clear":
+        builtins.__xonsh__.history = builtins.__xonsh__.history.__class__()
```
Comment (scopatz) [inline xonsh/history/main.py:424]: Doesn't this just get a new history object?

**Comment by scopatz** [comment]
Thanks for putting this in @Itay4! I agree with @laloch that this should be fleshed out a little more

**Pull request closed without merging**